src/preferences/dialog/dlgprefmixer.cpp

```diff
@@ -908,83 +991,85 @@ void DlgPrefMixer::slotMainEqEffectChanged(int effectIndex) {
-                    slider->setMaximum(static_cast<int>(pManifestParameter->getMaximum() * 100));
-                    slider->setSingleStep(1);
-                    slider->setValue(static_cast<int>(pManifestParameter->getDefault() * 100));
-                    slider->setMinimumHeight(90);
-                    // Set the index as a property because we need it inside slotUpdateFilter()
-                    slider->setProperty("index", QVariant(i));
-                    setScrollSafeGuard(slider);
-                    slidersGridLayout->addWidget(slider, 1, i + 1, Qt::AlignCenter);
-                    m_mainEQSliders.append(slider);
-                    // catch drag event
-                    connect(slider,
-                            &QSlider::sliderMoved,
-                            this,
-                            &DlgPrefMixer::slotApplyMainEQParameter);
-                    // catch scroll event
-                    connect(slider,
-                            &QSlider::valueChanged,
-                            this,
-                            &DlgPrefMixer::slotApplyMainEQParameter);
-
-                    QLabel* valueLabel = new QLabel(this);
-                    m_mainEQValues.append(valueLabel);
-                    QString valueText = QString::number((double)slider->value() / 100);
-                    valueLabel->setText(valueText);
-                    slidersGridLayout->addWidget(valueLabel, 2, i + 1, Qt::AlignCenter);
-
-                    // Immediately save the new (default) parameter values in preferences.
-                    // Otherwise, without pressing 'Reset parameter', the previously saved
-                    // parameter values (of another EQ effect) would be loaded on next start
-                    // which (unnoticed) messes up the parameters displayed now.
-                    m_pConfig->set(ConfigKey(kConfigGroup,
-                                           kMainEQParameterKey + QString::number(i + 1)),
-                            ConfigValue(valueText));
-                }
-            }
+    // Create and set up Main EQ's sliders
+    const auto params = pManifest->parameters();
+    for (const auto& param : params) {
+        if (param->parameterType() != EffectManifestParameter::ParameterType::Knob) {
+            continue;
         }
-    }
-
-    // Update the configured effect for the current QComboBox
-    if (pManifest) {
-        m_pConfig->set(ConfigKey(kConfigGroup, "EffectForGroup_[Master]"),
-                ConfigValue(pManifest->uniqueId()));
+        int i = param->index();
+
+        QLabel* pCenterFreqLabel = new QLabel(this);
+        QString labelText = param->name();
+        pCenterFreqLabel->setText(labelText);
+        m_mainEQLabels.append(pCenterFreqLabel);
+        slidersGridLayout->addWidget(pCenterFreqLabel, 0, i + 1, Qt::AlignCenter);
+
+        QSlider* pSlider = new QSlider(this);
```

Comment:
these are being leaked right now from what I can tell. seems like a not-insignificant memory leak to me.

Reply:
:+1: using parented_ptr for all spontaneously created widgets now, and for the numDecks ControlProxy.
no `new` anymore in this file.

Reply:
Thank you :+1:

Reply:
but I messed it up when using it for the Main EQ widgets,crashes with qDeleteAll and/or QList<Widged pointers> in `slotMainEqEffectChanged(int)` here https://github.com/mixxxdj/mixxx/pull/11527/commits/5f43d7e786fdce9c2474895f929ccf3a2d54ee24#diff-c4e587c957428bd6afd10136eb4dad00d554179984d800b2252947ef2c27e666L984

I'll take a look at this later on.
Or we stick to `new` and re-add the destructor to delete those widgets.

Reply:
I'll try to look into it later as well. You can probably just get rid of the `qDeleteAll` call.

Reply:
Looks like this is about some other widgets I made parented_ptr.

Reply:
Okay, I double checked, simply removing the `qDeleteAll`s is the correct choice. The compiler should've warned you if you were to call `qDeleteAll` on a `QList<parented_ptr<T>>`, but it apparently didn't. The resulting code still won't be leaky since all objects in the QList will have a parent (as `DEBUG_ASSERT`ed when calling `.clear()` on the container).

The thing I'm rather confused by is that apparently `QList::append`ing a `parented_ptr<T>` works, even though it shouldn't because copy and move have been deleted...

Reply:
Then its probably from [this lambda](https://github.com/mixxxdj/mixxx/pull/11527/files#diff-c4e587c957428bd6afd10136eb4dad00d554179984d800b2252947ef2c27e666R226-R235). It's probably that the lambda (the connection its bound to) outlives the `pQuickEffectComboBox` reference, which lives shorter because of some `m_deckQuickEffectSelectors.clear()` call. But I wonder why that bug wasn't there previously.

Reply:
It's not the Main EQ (and therefore: false alarm, qDeleteAll doesn't cause any issues), and there is no `m_deckQuickEffectSelectors.clear()`.

But it's indeed that lambda and that `pQuickEffectComboBox` reference, though the crash happens during startup and the presetChanged() signal is not emitted.

I worked around it by capturing just the deck group string and the preset name and update the index in a separate slot (just to have a compact lambda body).

Reply:
...and I added a note about the crash and the workaround.